aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,213 @@
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .keyType(KeyType.HASH)
+          .build()
+  );
+
+  private static final List<AttributeDefinition> LOCK_TABLE_COL_DEFINITIONS = Lists.newArrayList(
+      AttributeDefinition.builder()
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .attributeType(ScalarAttributeType.S)
+          .build()
+  );
+
+  private DynamoDbClient dynamo;
+  private String lockTableName;
+
+  /**
+   * constructor for dynamic initialization, {@link #initialize(Map)} must be called later.
+   */
+  DynamoLockManager() {
+  }
+
+  /**
+   * All arg constructor, used for testing purpose
+   * @param dynamo dynamo client
+   * @param lockTableName lock table name
+   */
+  DynamoLockManager(DynamoDbClient dynamo, String lockTableName) {
+    this.dynamo = dynamo;
+    this.lockTableName = lockTableName;
+    ensureLockTableExistsOrCreate();
+  }
+
+  private void ensureLockTableExistsOrCreate() {
+    ensureTableExistsOrCreate(lockTableName, LOCK_TABLE_SCHEMA, LOCK_TABLE_COL_DEFINITIONS);
+  }
+
+  private void ensureTableExistsOrCreate(String tableName, List<KeySchemaElement> schema,
+                                         List<AttributeDefinition> definitions) {
+    if (tableExists(tableName)) {
+      return;
+    }
+
+    LOG.info("Dynamo lock table {} not found, trying to create", tableName);
+    dynamo.createTable(CreateTableRequest.builder()
+        .tableName(tableName)
+        .keySchema(schema)
+        .attributeDefinitions(definitions)
+        .billingMode(BillingMode.PAY_PER_REQUEST)
+        .build());
+
+    Tasks.foreach(tableName)
+        .retry(LOCK_TABLE_CREATION_WAIT_ATTEMPTS_MAX)
+        .throwFailureWhenFinished()
+        .run(this::checkTableActive);
+  }
+
+  private boolean tableExists(String tableName) {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
+      return true;
+    } catch (ResourceNotFoundException e) {
+      return false;
+    }
+  }
+
+  private void checkTableActive(String tableName) {
+    try {
+      DescribeTableResponse response = dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
+      TableStatus currentStatus = response.table().tableStatus();
+      if (!currentStatus.equals(TableStatus.ACTIVE)) {
+        throw new IllegalStateException(String.format("Dynamo table %s is not active, current status: %s",
+            tableName, currentStatus));
+      }
+    } catch (ResourceNotFoundException e) {
+      throw new IllegalStateException(String.format("Cannot find Dynamo table %s", tableName));
+    }
+  }
+
+  @Override
+  public void initialize(Map<String, String> properties) {
+    this.dynamo = AwsClientFactories.from(properties).dynamo();
+    this.lockTableName = properties.get(AwsProperties.LOCK_DYNAMO_TABLE);
+    Preconditions.checkNotNull(lockTableName, "tableName must not be null");
```

Comment:
It is better to avoid including class, method, or variable names in messages because the user often won't be familiar with this code's variable names. Instead, use "Table name" or "Lock table name" because that's what the user sets.